### PR TITLE
now with pasta

### PR DESCRIPTION
**Pull request by niftypsycho**
added new method 'pastafy' and trigger word for calling it 'slurp'.  Now imports 'emoji' at top.  slurped comments get emojis added wherever there is a word that is an emoji alias.

### CODE REVIEW

**Comment by crypticism** [comment]
What does this actually do?

File: dxbot2.py [hunk 1]
```diff
@@ -69,6 +71,36 @@ def spongeify(message):
...
+					if couldbemoji in custom_emoji_list:
+						splitnoodle[findex] = (grain+couldbemoji)
+			pasta[index] = (':').join(splitnoodle)
+		else:
+			couldbemoji = ':{}:'.format(noodle)
+			if couldbemoji in emoji.EMOJI_ALIAS_UNICODE:
+				pasta[index] = (noodle+couldbemoji)
+			if couldbemoji in custom_emoji_list:
+					pasta[index] = (couldbemoji)
+	return u_temp+' '.join(pasta)
```
[the remark this answers is not in the document]
Comment (tomatosoupcan) [reply dxbot2.py]: Also having emoji already in the message seems like it causes issues, maybe strip all : from the string before starting or something

[the remark this answers is not in the document]
Comment (niftypsycho) [reply dxbot2.py]: There is a command to convert ':text:' into a unicode string, but I think slack expects ':text:' and converts it internally, so to get the emoji you just pass it surrounded by colons.

[the remark this answers is not in the document]
Comment (niftypsycho) [reply dxbot2.py]: I'm trying to keep to the standard set by r/emojipasta where the word is followed by its applicable emoji - both appear in the text.  There is a check if there is 2 or more colons - there could be an emoji in the word.  It should keep any emoji that it's passed while also adding any to the word.  I tried to keep message integrity, but if the word is complicated like ':herb:bacon::man_boy' the code will turn them all into emojis, keeping the ones originally written and adding new ones.

File: dxbot2.py [hunk 2]
```diff
@@ -69,6 +71,36 @@ def spongeify(message):
     ]
     return u_temp+''.join(sponge)
 
+def pastafy(message):
+	emoji = client.api_call('emoji.list')['emoji']
```
[the remark this answers is not in the document]
Comment (niftypsycho) [reply dxbot2.py]: Fixed

File: dxbot2.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomatosoupcan) [reply dxbot2.py]: Have you tested the emoji at the end of the string? I had issues with that

File: dxbot2.py [hunk 3]
```diff
@@ -69,6 +71,32 @@ def spongeify(message):
     ]
     return u_temp+''.join(sponge)
 
+def pastafy(message):
+    emoji = client.api_call('emoji.list')['emoji']
+    custom_emoji_list = emoji.keys()
+
```
Comment (tomatosoupcan) [inline dxbot2.py]: This line has an indent in it that does not need to be there

File: dxbot2.py [hunk 4]
```diff
@@ -69,6 +71,32 @@ def spongeify(message):
...
+    pasta = m_temp.split()
+    for index, noodle in enumerate(pasta):
+        if noodle.count(':') > 1:
+	    splitnoodle = noodle.split(':')
+	    for findex, grain in enumerate(splitnoodle):
+	        if grain:
+		    couldbemoji = ':{}:'.format(grain)
+		    if couldbemoji in emoji.EMOJI_ALIAS_UNICODE or couldbemoji in custom_emoji_list:
+		        splitnoodle[findex] = (grain+couldbemoji)
+	# Removes colons
```
Comment (tomatosoupcan) [inline dxbot2.py]: Fails on the following test case:

"king_buttons :+1 heart thermometer:"

Returns :+1 heart❤️  thermometer:

File: dxbot2.py [hunk 5]
```diff
@@ -69,6 +71,32 @@ def spongeify(message):
...
+	pasta[index] = ('').join(splitnoodle)
```
[the remark this answers is not in the document]
Comment (niftypsycho) [reply dxbot2.py]: If you're talking about the comment, I changed it so it's more obvious it's talking about the line below it.

File: dxbot2.py [hunk 6]
```diff
@@ -209,10 +241,12 @@ def handle_command(command, args, channel, prev):
         else:
             response = 'No arguments provided'
 
-    if command.startswith(('grab','yoink','snag')):
+    if command.startswith(('grab','yoink','snag','slurp')):
         message = '{} {}'.format(user_map[prev['user']], prev['text'])
         if command.startswith('yoink'):
             message = spongeify(message)
+		if command.startswith('slurp'):
```
[the remark this answers is not in the document]
Comment (niftypsycho) [reply dxbot2.py]: Responding to this for completion.  Indent problems have been addressed.

[the remark this answers is not in the document]
Comment (niftypsycho) [reply dxbot2.py]: No longer indented too far.

[the remark this answers is not in the document]
Comment (tomatosoupcan) [reply dxbot2.py]: This still looks wrong, maybe I just can't see the most recent changes?

File: dxbot2.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (tomatosoupcan) [reply dxbot2.py]: I mean the line following the comment should be indented one further

File: dxbot2.py (hunk 4 above)
Comment (tomatosoupcan) [reply dxbot2.py]: Did you resolve this? I can't tell.

**Pull request merged**